### DISCUSSION TITLE
Dropdown Menu - Preventing default Portal render

### DISCUSSION THREAD

**Discussion by exponent28** [shadcn-ui/ui#1003] [Q&A]
Hello, 

Does anyone know how to render the dropdown menu content inside the normal parent div (no portal at all)? 

Have tried to override with portal custom container though this still portals the content outside of the document.body hierarchy. 

Thanks :))

**Comment by Moicky** [comment]
Just faced the same issue when using tiptap's editor while putting a Dropdown inside their BubbleMenu.

I was able to solve it by passing the `container` prop to the Portal which <DropdownMenuContent/> uses:
https://www.radix-ui.com/primitives/docs/components/dropdown-menu#portal

**Reply by Moicky** [reply]
Looks something like this:
![Screenshot 2024-02-03 at 15 12 46](https://github.com/shadcn-ui/ui/assets/77806634/6dd5097f-793a-485b-987d-21afc98bbbe4)

**Comment by Maliksidk19** [comment]
![image](https://github.com/user-attachments/assets/97740415-01f1-41d6-84af-75776bffd682)

use this

**Reply by halid96** [reply]
bro you saved my day, thank you so much!

**Comment by Langlw1992** [comment]
For those troubled by this issue, you can try the following solution  1.  set the modal property to false
<img width="720" height="334" alt="image" src="https://github.com/user-attachments/assets/72fd17a4-09e0-443e-83b5-cbda82b58b81" />
2. remove the default portal wrapper in `components/ui/dropdown-menu.tsx` 
<img width="1400" height="892" alt="image" src="https://github.com/user-attachments/assets/1b707b8c-5b40-444f-8325-e1df6433dcbf" />
it works in V4
